pkg/aws/ec2/api/helper.go

```diff
@@ -60,17 +63,34 @@ var (
 )
 
 type ec2APIHelper struct {
-	ec2Wrapper EC2Wrapper
+	ec2Wrapper                 EC2Wrapper
+	workerNodeVpcId            string
+	securityGroupNameToIdCache cache.Store
 }
 
-func NewEC2APIHelper(ec2Wrapper EC2Wrapper, clusterName string) EC2APIHelper {
+type securityGroupToId struct {
+	securityGroupName string
+	securityGroupId   string
+}
+
+func NewEC2APIHelper(ec2Wrapper EC2Wrapper, clusterName, workerNodeVpcId string) EC2APIHelper {
 	// Set the key and value of the cluster name tag which will be used to tag all the network interfaces created by
 	// the controller
 	clusterNameTag = &ec2.Tag{
 		Key:   aws.String(fmt.Sprintf(config.ClusterNameTagKeyFormat, clusterName)),
 		Value: aws.String(config.ClusterNameTagValue),
 	}
-	return &ec2APIHelper{ec2Wrapper: ec2Wrapper}
+	securityGroupCacheKeyFunc := func(obj interface{}) (string, error) {
+		securityGroup := obj.(*securityGroupToId)
+		return securityGroup.securityGroupName, nil
+	}
+
+	securityGroupNameToIdCache := cache.NewTTLStore(securityGroupCacheKeyFunc, 15*time.Minute)
```

Comment:
Note: The TTL duration could be made configurable by the user based on their use-case. 

I'll discuss this internally with the team.

Reply:
@vgrigoruk  did this get discussed internally?